airbyte-integrations/connectors/destination-weaviate/destination_weaviate/destination.py

```diff
@@ -0,0 +1,94 @@
+        successfully persisted to the destination. This is used to ensure fault tolerance in the case that a sync fails before fully completing,
+        then the source is given the last state message output from this method as the starting point of the next sync.
+
+        :param config: dict of JSON configuration matching the configuration declared in spec.json
+        :param configured_catalog: The Configured Catalog describing the schema of the data being received and how it should be persisted in the
+                                    destination
+        :param input_messages: The stream of input messages received from the source
+        :return: Iterable of AirbyteStateMessages wrapped in AirbyteMessage structs
+        """
+        client = Client(config, get_schema(configured_catalog))
+        # TODO add support for overwrite mode
+        # for configured_stream in configured_catalog.streams:
+        #    if configured_stream.destination_sync_mode == DestinationSyncMode.overwrite:
+        #        client.delete_stream_entries(configured_stream.stream.name)
+
+        for message in input_messages:
+            if message.type == Type.STATE:
+                # Emitting a state message indicates that all records which came before it have been written to the destination. So we flush
+                # the queue to ensure writes happen, then output the state message to indicate it's safe to checkpoint state
+                client.flush()
+                yield message
+            elif message.type == Type.RECORD:
+                record = message.record
+                client.queue_write_operation(record.stream, record.data)
+            else:
+                # ignore other message types for now
+                continue
+
+        # Make sure to flush any records still in the queue
+        client.flush()
+
+    def check(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> AirbyteConnectionStatus:
+        """
+        Tests if the input configuration can be used to successfully connect to the destination with the needed permissions
+            e.g: if a provided API token or password can be used to connect and write to the destination.
+
+        :param logger: Logging object to display debug/info/error to the logs
+            (logs will not be accessible via airbyte UI if they are not passed to this logger)
+        :param config: Json object containing the configuration of this destination, content of this json is as specified in
+        the properties of the spec.json file
+
+        :return: AirbyteConnectionStatus indicating a Success or Failure
+        """
+        try:
+            client = Client.get_weaviate_client(config)
+            ready = client.is_ready()
+            class_name = ''.join(random.choices(string.ascii_uppercase, k=10))
+            client.schema.create_class({"class": class_name})
```

Comment:
can't you use the queue_write_operation() method for creating a class and writing a record simultaneously?

Reply:
that wouldn't work if the weaviate cluster was configured with dynamic schema creation disabled. The check to create an emtpy class is enough because that is considered a write operation as well in Weaviate so it ensures we don't have read-only access. There is no granular access yet so having permission to write a class means you also have permission to write an object.

Reply:
if it wouldn't work in the **_check()_** operation wouldn't that mean that you won't be able to perform the same operation on **_write()_** as well? in that scenario if for some reason the user forgets to pr-create a class and schema the connector will fail the replication?